### ISSUE TITLE
Regolith | Winterdev

### ISSUE THREAD

**Issue by Magnogen** [Magnogen/OpenMusic#2]
Hey Winter, I've gotten started with a shortish main menu track for your game, I've included the file below

[Walking On Sound](../blob/main/music/background/ambient_chill/WalkingOnSound.wav)

**Comment by IainWinter** [comment]
Nice this sounds good! Getting some Lifeformed vibes. Though, I don't think that the main menu music should be this melancholic. I'm telling you that Radius song you made is near perfect vibes, I don't know how you feel about touching up a past song? I've was initially thinking that a song for the main menu should have a chill drum shuffle with some synths which is exactly what Radius is by coincidence lol. 

Not to be pushy cus you are doing this for free! but I think that if we start on the battle music first, that will be easier to place the feeling because it should just be upbeat. The main menu is a much more complex feel that I think Radius already almost has.

Let me know what you think about working with Radius again.

I love those PhyTrance songs that you have and I saw in the description that you were wanting to make some more of those, so maybe that'd be a good place to explore.

On the actual sound effect side of things I think we need:

Gameplay:
Main gun - shoots every second
Minigun - shoots every 1/10 seconds
Big Laser - continuous
Small Laser - every second, but multiple are happening at once
Electricity - continuous
Electricity ball - continuous, maybe has another effect layered ontop idk
Movement - small thruster noise
explosion - for enemy ships
when a cell gets hit by a bullet a sound should play based on the material - right now there are metal, rock, gas
player hit - big metal clash
player death timer - some alarm
player hit core - alien sounding metal clash
player death - some explosion with reverb. could also be a slash noise something in the high register ontop of the regular explosion sound

Menus:
Button clicking - The menus are going to have an industrial feel to them, a web of pipes and wires, so maybe this should be a sample of a real button on some machine with a nice click, i might be able to record one
----   these need a hover and click sound (hover rising and clicking falling past where the rise started)
Ambient - Quiet background sounds of steam and rumbling machines
Typing - small click

I am excited to hear what you have to make! 

I am reprogramming the sound engine, I'll let you know when I have that not broken :)

**Comment by Magnogen** [comment]
Nice! Thanks for giving such a detailed description of what you're looking for! I'd be happy to provide, shall I make multiple sounds of each - to add more variation to each sound? Kind of similar to what Minecraft does I suppose.

**Comment by IainWinter** [comment]
Yeah that would prob be a good idea. I think that making a rough draft of a bunch of them and then perfecting it would be the way to go. So don't worry about getting the prefect sound in one shot btw

**Comment by Magnogen** [comment]
Hey @IainWinter, I've uploaded a synthwave track that I think you might find handy. I've got a techno track that I'll be uploading later this evening, as well as some click sounds. 

[Misfortune](https://magnogen.net/OpenMusic/music/background/upbeat/01_Misfortune.wav)

**Comment by Magnogen** [comment]
Okay, Terra Firma took longer than I thought it would, sorry about that. I've uploaded it now as well as other sound effects. Hope the gamedev is going well!

[Terra Firma](https://magnogen.net/OpenMusic/music/background/upbeat/02_TerraFirma.wav)

[Gunshots](https://github.com/Magnogen/OpenMusic/tree/main/soundfx/futuristic/gunshots)

[Clicks](https://github.com/Magnogen/OpenMusic/tree/main/soundfx/realistic/clicks)

**Comment by IainWinter** [comment]
Nice! Let me give you a build I forgot to do that.

Here it is there are some sounds that I made and a song from you, if you press escape you can change the volume.

[build](https://winter.dev/engine/build.zip)

These sounds effects are good, but like I was saying I really want to try and get away from the 8 bit sound. I want this to be a more modern sounding game. I am using this audio engine called FMOD, which allows you to have synths at runtime so I actually need to construct the sound effects closer to the code, so let me worry about those. I found a plugin called Weaponizer that was only $100 and comes with tons of sounds for this so I think that I can use those for the effects. 

On the music side of things, Terra Firma is my fav so far. I liked the first two, but I don't think they really fit the space vibe. I'm down to put really any type of music into this, it just can't be these retro synths!!! That house song you have, Trust could also fit because it has only modern synths and a heavy DnB thread. Texture / Texture VIP are other ones that aren't particularly spacey but have an amazing meandering feel that works well for background music. 

My brother is a drummer, so if you make a song with realistic drums, later we can mix in a real recording which could be cool. We could also mix in some real guitars which would be cool.

I mixed some of your old songs with the new ones and I think you will agree with what I'm saying while playing :)

Let me know what you think, thanks for all this!!

**Comment by Magnogen** [comment]
Awesome okay! I've actually released an EP a little while back, I just haven't been able to release it on my YouTube Channel. Its more on the Dubstep side, with some PsyTrance and DnB. You can find it here:

[The Cube EP - Ghost Records](https://www.youtube.com/playlist?list=PLTbKSu6fTzqq3wJj5v6-9gl50ov4o40Mt)

You might like some of them, perhaps Eight Vertices the most, as it has more realistic drums and a DnB vibe.

I'm looking throught the build now, and will let you know what I think! :P

**Comment by Magnogen** [comment]
Hmm, it appears to break on my machine, and I'm not sure how I can share a crash report.

**Comment by IainWinter** [comment]
That's my bad I built it with the wrong asset path. It should print the error to a filed named 'CONERR$' in the same folder as the exe. I just updated the one from the same link so try and redownload it.

Every time you run it, it should play a random song btw

The main gun might sound a little weird for a gun but in lore the main character is going to be an alien ship that is more throwing the orange pixels instead of shooting a gun :)

I'll listen to these thanks!

**Comment by Magnogen** [comment]
Damn that's pog, I noticed that sometimes the shooting sfx line up with the beats of the music lol

I'll see what else I can make that fits with this vibe, I think I might redo Terra Firma as the rythm feels a bit off.

A few ideas for the gameplay itself too:
- The ship itself seems to not have very much health (i.e. I die a lot oof)
- Bullets almost _always_ miss their target, 
  - Perhaps just a little bit of aimassist to help out?
  - Or making the smaller enemies move slower could work too
  - The bomb things work pretty well though, as they dont really get too close the the player
- The controls are pretty obvious, unless there's more than just WASD and Left Click
  - An instruction screen could work well
  - Or you could do some sort of skippable tutorial section, before actual enemies appear
  - I personally feel that WASD based on the coordinates is a bit hard to control
    - maybe make the ship rotate towards the mouse and move forward when W is pressed?
    - and I s'pose the same for ASD, just relative to the direction to the mouse

**Comment by IainWinter** [comment]
Sweet you got it working! I think just seeing the game works much better than me trying to describe feelings over this chat lol. 

I think I need to add an I-frames after each hit to the core to stop random instant death, cus that def happens a lot. The upgrades will solve the aim problems I am adding an aim assist that shows you a target and magnetic bullets that will curve towards enemies. I like the bare bones start that you get and over time you will upgrade the ship to be better, you'll see in the next build :)

I tried different movement systems, and this seemed to be the best by far lol. Some upgrades are going to involve the control of the ship, like its acceleration and stuff. I think that it'll control more fluidly at the start and then what it is now might be the ultimate upgraded version because it's perfect stop start so you have absolute control if that makes sense.

Thanks for the feedback. I'll probs include a Binding of Isaac style tutorial that fades out after sometime at the start.
